torch/onnx/errors.py

```diff
@@ -5,7 +5,7 @@
 from typing import Optional
 
 from torch import _C
-from torch.onnx import _constants
+from torch.onnx import _constants, diagnostic
```

Comment:
`diagnostics`?

Reply:
I was going in circle about this as well for a while. I feel singular better depicts the system (e.g. DiagnosticEngine, DiagnosticSystem), while plural often represents actual "diagnostics". See https://github.dev/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs .

Reply:
c# likes to name their files according to the class names, which imo is different than how we typically name python modules (usually a collection of types and functions). Happy to discuss further.

Reply:
Ah I should send a shorter link cause I was referring the folder name. But then I noticed the namespace is plural, hmm...

Reply:
😉